tests/foreman/api/test_http_proxy.py

```diff
@@ -125,3 +126,75 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
     )
     assert len(yum_repo['output']) >= 1
     assert 'quay/busybox' in yum_repo['output']
+
+
+@pytest.mark.upgrade
+@pytest.mark.rhel_ver_match('8')
+@pytest.mark.run_in_one_thread
+@pytest.mark.parametrize(
+    'setup_http_proxy',
+    [None, True, False],
+    indirect=True,
+    ids=['no_http_proxy', 'auth_http_proxy', 'unauth_http_proxy'],
+)
+@pytest.mark.tier3
+def test_positive_auto_attach(
+    setup_http_proxy, module_target_sat, rhel_contenthost, function_entitlement_manifest_org
+):
+    """Attempt to auto attach a subscription to content host
+
+    :id: cce888b5-e023-4ee2-bffe-efa9260224ee
+
+    :customerscenario: true
+
+    :expectedresults: host successfully subscribed, subscription
+        repository enabled, and repository package installed.
+
+    :Assignee: jpathan
+
+    :BZ: 2046337
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    org = function_entitlement_manifest_org
+    lce = module_target_sat.api.LifecycleEnvironment(organization=org).create()
+    content_view = module_target_sat.api.ContentView(organization=org).create()
+    rh_repo_id = enable_sync_redhat_repo(constants.REPOS['rhel8_bos'], org.id)
```

Comment:
why enable large rhel8_bos repo? As an alternative, would recommend enabling an infra_podman custom repo, or if a CDN repo is needed, try the ansible2.9 repo for auto-attach and install any package from that repo?

Reply:
Historically there have been issues in syncing large repos via HTTP proxy, that's why I am testing rhel8_bos.